crates/rmf_site_editor/src/site/save.rs

```diff
@@ -1440,31 +1449,39 @@ fn generate_scenarios(
                                             },
                                         ))
                                     })
-                                    .collect(),
+                                    .collect::<Result<_, _>>()?,
                                 tasks: scenario_task_modifiers
                                     .iter()
-                                    .filter_map(|child_entity| {
-                                        task_modifiers.get(*child_entity).ok()
-                                    })
-                                    .filter_map(|(inclusion, task_params, affiliation)| {
-                                        Some((
-                                            affiliation.0.and_then(|e| tasks.get(e).ok())?.0,
+                                    .map(|e| {
+                                        let (inclusion, task_params, affiliation) = task_modifiers
+                                            .get(*e)
+                                            .map_err(|_| SiteGenerationError::BrokenModifier(*e))?;
+                                        let a = affiliation
+                                            .0
+                                            .ok_or(SiteGenerationError::EmptyAffiliation(*e))?;
+                                        Ok((
+                                            tasks.get(a).map(|id| id.0).map_err(|_| {
+                                                SiteGenerationError::BrokenAffiliation(a)
+                                            })?,
```

Comment:
Same as above about bring the logic outside of the Ok statement to make it a bit easier to follow